client/my-sites/stats/post-trends/week.jsx

```diff
@@ -55,19 +53,26 @@ export default React.createClass( {
 			days.push(
 				<Day key={ dayDate.format( 'MMDD' ) }
 					className={ classNames.join( ' ' ) }
-					label={ dayDate.format( 'L' ) }
+					label={ dayDate.locale( this.props.userLocale ).format( 'L' ) }
 					postCount={ postCount }
 				/>
 			);
 		}
 
 		return days;
-	},
+	}
 
 	render() {
 		return (
 			<div className="post-trends__week">{ this.getDayComponents() }</div>
 		);
 	}
 
-} );
+}
+
+export default connect(
+	( state ) => ( { userLocale: getCurrentUserLocale( state ) } ),
+	{},
+	null,
+	{ pure: false }
```

Comment:
For my own knowledge, I'm wondering what is the exact rule to add the `pure: false` option? I'm not sure I understand it correctly